Testing for invalid hierarchical JSON path provokes crash

If the JSON path in a call to `hasJson()` includes a hierarchical path which doesn't exist then we get a crash and a stack trace that looks like:

```
(venv) ➜  core git:(2.2) ✗ python3 -m pytest -k test_multi_json_response -s
======================================================================================================= test session starts =======================================================================================================
platform darwin -- Python 3.7.4, pytest-4.6.2, py-1.8.0, pluggy-0.12.0
rootdir: /Users/allancochrane/work/core, inifile: pytest.ini
plugins: cov-2.7.1
collecting 0 items                                                                                                                                                                                                                Cannot find controller Controller. Did you create this one?
Cannot find controller Controller. Did you create this one?
collected 454 items / 453 deselected / 1 selected

tests/testing/test_route_tests.py 4437275152
GET Route: /unit/test/json/multi
4437275152
GET Route: /unit/test/json/multi
JSON multi =  {"author": {"name": "Joe"}}
4437275152
GET Route: /unit/test/json/multi
4437275152
GET Route: /unit/test/json/response
JSON response =  {"count": 5, "iterable": [1, 2, 3]}
4437275152
GET Route: /unit/test/json/response
4437275152
GET Route: /unit/test/json/response
4437275152
GET Route: /unit/test/json/response
F

============================================================================================================ FAILURES =============================================================================================================
______________________________________________________________________________________________ TestUnitTest.test_multi_json_response ______________________________________________________________________________________________

self = <tests.testing.test_route_tests.TestUnitTest testMethod=test_multi_json_response>

    def test_multi_json_response(self):
        self.assertTrue(self.json('GET', '/unit/test/json/multi').hasJson({
            'author.name': 'Joe'
        }))

        print("JSON multi = ", self.json('GET', '/unit/test/json/multi').container.make('Response'))
        self.assertTrue(self.json('GET', '/unit/test/json/multi').hasJson('author.name', 'Joe'))
        # Crashes: self.assertFalse(self.json('GET', '/unit/test/json/multi').hasJson('author.name.foo', 'foo'))
        print("JSON response = ", self.json('GET', '/unit/test/json/response').container.make('Response'))
        self.assertFalse(self.json('GET', '/unit/test/json/response').hasJson('password.foo', 'foo'))
        self.assertFalse(self.json('GET', '/unit/test/json/response').hasJson('foo.foo', 'foo'))
>       self.assertFalse(self.json('GET', '/unit/test/json/response').hasJson('count.foo', 'foo'))

tests/testing/test_route_tests.py:85:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
masonite/testing/MockRoute.py:44: in hasJson
    return Dot().dot(key, response, False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <masonite.helpers.structures.Dot object at 0x10b349850>, search = 'count.foo', dictionary = {'count': 5, 'iterable': [1, 2, 3]}, default = False

    def dot(self, search, dictionary, default=None):
        """The search string in dot notation to look into the dictionary for.

        Arguments:
            search {string} -- This should be a string in dot notation
                                like 'key.key.value'.
            dictionary {dict} -- A normal dictionary which will be searched using
                                the search string in dot notation.

        Keyword Arguments:
            default {string} -- The default value if nothing is found
                                in the dictionary. (default: {None})

        Returns:
            string -- Returns the value found the dictionary or the default
                        value specified above if nothing is found.
        """
        if '.' not in search:
            if search == '':
                return dictionary
            try:
                return dictionary[search]
            except KeyError:
                return default

        searching = search.split('.')
        possible = None
        while searching:
            dic = dictionary
            for value in searching:
                if not dic:
                    if '*' in searching:
                        return []
                    return default

                if isinstance(dic, list):
                    try:
                        return collect(dic).pluck(searching[searching.index('*') + 1]).serialize()
                    except KeyError:
                        return []
>               dic = dic.get(value)
E               AttributeError: 'int' object has no attribute 'get'

masonite/helpers/structures.py:50: AttributeError
============================================================================================ 1 failed, 453 deselected in 1.05 seconds =============================================================================================
```


**To Reproduce**
Steps to reproduce the behavior:
Add this line to the test named test_multi_json_response in file test_route_tests.py:

    self.assertFalse(self.json('GET', '/unit/test/json/response').hasJson('count.foo', 'foo'))

**Expected behavior**
I expected `hasJson()` to return `False`

**Additional context**
Git log:

commit 94a20b713d179d4bde79f0d313ed6c5bba2676f4 (HEAD -> 2.2, origin/HEAD, origin/2.2)
Author: Joseph Mancuso <idmann509@gmail.com>
Date:   Sat Sep 7 17:53:28 2019 -0400

    fixed issue with config helper not getting False values

--- a/app/http/controllers/UnitTestController.py
+++ b/app/http/controllers/UnitTestController.py
@@ -36,7 +36,7 @@ class UnitTestController(Controller):
     def response(self):
         return {
             'count': 5,
-            'iterable': [1,2,3]
+            'iterable': [1, 2, 3]
         }
 
     def multi(self):
@@ -45,6 +45,9 @@ class UnitTestController(Controller):
                 'name': 'Joe'
             }
         }
+
+    def multi_count(self):
+        return {"count": 5, "iterable": [1, 2, 3]}
 
     def patch(self):
         return self.request.input('test')

--- a/masonite/helpers/structures.py
+++ b/masonite/helpers/structures.py
@@ -47,8 +47,12 @@ class Dot:
                         return collect(dic).pluck(searching[searching.index('*') + 1]).serialize()
                     except KeyError:
                         return []
-                dic = dic.get(value)
+                
+                if not isinstance(dic, dict):
+                    return default
 
+                dic = dic.get(value)
+                
                 if isinstance(dic, str) and dic.isnumeric():
                     continue
 

--- a/masonite/testing/TestCase.py
+++ b/masonite/testing/TestCase.py
@@ -211,7 +211,6 @@ class TestCase(unittest.TestCase):
         wsgi.update(wsgi_values)
         self.container.bind('Environ', wsgi)
         self.container.make('Request')._test_user = self.acting_user
-        print(id(self.container))
         self.container.make('Request').load_app(self.container)
         if self._with_subdomains:
             self.container.make('Request').activate_subdomains()

--- a/routes/web.py
+++ b/routes/web.py
@@ -31,6 +31,7 @@ ROUTES = [
         Get('/test/json/response', 'UnitTestController@response'),
         Post('/test/json/validate', 'UnitTestController@validate'),
         Get('/test/json/multi', 'UnitTestController@multi'),
+        Get('/test/json/multi_count', 'UnitTestController@multi_count'),
         Patch('/test/patch', 'UnitTestController@patch'),
     ], prefix="/unit")
 ]

--- a/tests/helpers/test_dot_notation.py
+++ b/tests/helpers/test_dot_notation.py
@@ -23,6 +23,7 @@ class TestDot(unittest.TestCase):
         self.assertEqual(DictDot().dot('key.test.layer', {'key': {'test': {'layer': 'value'}}}), 'value')
         self.assertEqual(DictDot().dot('key.none', {'key': {'test': {'layer': 'value'}}}), None)
         self.assertEqual(DictDot().dot('key', {'key': {'test': {'layer': 'value'}}}), {'test': {'layer': 'value'}})
+        self.assertEqual(DictDot().dot('key.test.none', {'key': 'value'}), None)
 
     def test_dict_dot_asterisk(self):
         payload = {

--- a/tests/testing/test_route_tests.py
+++ b/tests/testing/test_route_tests.py
@@ -77,6 +77,7 @@ class TestUnitTest(TestCase):
         }))
 
         self.assertTrue(self.json('GET', '/unit/test/json/multi').hasJson('author.name', 'Joe'))
+        self.assertFalse(self.json('GET', '/unit/test/json/multi_count').hasJson('count.foo', 'foo'))
 
     def test_count(self):
         self.assertTrue(self.json('GET', '/unit/test/json/response').count(2))

